### DISCUSSION TITLE
Multi Zones with Common Components shared by both the Zones

### DISCUSSION THREAD

**Discussion by ps011** [vercel/next.js#25023] [Help]
We have this requirement of using multi zones in our project. I wanted to understand if we can have a components directory shared between both the zones ?

**Comment by sergioviniciuss** [comment]
I'm also researching on the same topic, but the best thing I can think of so far is creating an NPM package and importing it in the zones.

e.g.: you create a header component version 1.0.0 and install it in both zones. in the zones, you fetch the data required and pass it to the header, imported from the npm module, as props.

This approach requires a bit of coordination if you need to make sure the versions are aligned between the zones, but since webpack modules federation is not supported in nextjs natively, it feels this is the price to pay... (and we can hope in the future nextjs supports it). If they do, we'll be able to expose shared components and consume it in the zones.

**Reply by ps011** [reply]
I also had this in my mind but I thought to keep this as the last option for the reasons that you already mentioned in your answer about version management.

**Reply by sergioviniciuss** [reply]
it feels this is the best approach... I'd like to see nextjs team recommendation for this. With multi-zones, this question could be in their docs (or in the FAQ)
